### PR TITLE
Gather operator

### PR DESCRIPTION
**Pull request by scxiao**
the axis attribute in the onnx spec is with the int type, but previous implementation use size_t that makes it cannot be negative values. Change to int to support negative axis input.

### CODE REVIEW

File: src/include/migraphx/operators.hpp [hunk 1]
```diff
@@ -635,17 +635,25 @@ struct as_shape
 
 struct gather
 {
-    std::size_t axis = 0;
+    mutable int axis = 0;
```
Comment (pfultz2) [inline src/include/migraphx/operators.hpp]: Let not use `mutable`. We should just recompute the axis everytime we need it(it should be just a simple subtraction).

File: test/op_shape_test.cpp [hunk 2]
```diff
@@ -227,7 +227,7 @@ TEST_CASE(gather)
     {
         migraphx::shape input{migraphx::shape::float_type, {2, 3, 4, 5}};
         migraphx::shape indices{migraphx::shape::int32_type, {2, 3}};
-        std::size_t axis = 4;
+        int axis = 4;
         throws_shape(migraphx::op::gather{axis}, input, indices);
     }
```
Comment (pfultz2) [inline test/op_shape_test.cpp]: Can we add a test with a negative axis here, instead of a gpu test?
